src/Nancy/IO/RequestStream.cs

```diff
@@ -358,9 +358,23 @@ private void ShiftStreamToFileStreamIfNecessary()
 
         private static FileStream CreateTemporaryFileStream()
         {
-            var filePath = Path.GetTempFileName();
-
-            return new FileStream(filePath, FileMode.Create, FileAccess.ReadWrite, FileShare.None, 8192, StaticConfiguration.AllowFileStreamUploadAsync);
+            try
+            {
+                var filePath = Path.GetTempFileName();
+
+                return new FileStream(
+                    filePath,
+                    FileMode.Create,
+                    FileAccess.ReadWrite,
+                    FileShare.None,
+                    8192,
+                    StaticConfiguration.AllowFileStreamUploadAsync);
+            }
+            catch (IOException ex)
+            {
+				throw new IOException(
+                        "Unable to create temporary file, the %TEMP% folder has more than 65535 items. Try clearing out the folder.", ex);

```

Comment:
https://msdn.microsoft.com/en-us/library/system.io.path.gettempfilename(v=vs.110).aspx

"The GetTempFileName method will raise an IOException if no unique temporary file name is available. To resolve this error, delete all unneeded temporary files."